File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        endResult = []
        for x in range(0, len(nums)):
            result = 1
            for y in range(0, len(nums)):
                if x == y:
                    continue
                else:
                    result *= nums[y]
            endResult.append(result)
        return endResult

File: Data Structures & Algorithms/permutations/submission-0.java
class Solution {
    public List<List<Integer>> permute(int[] nums) {
        List<List<Integer>> result = new ArrayList<>();
        backtrack(nums, new ArrayList<>(), result, new boolean[nums.length]);
        return result;
    }

    private void backtrack(int[] nums, List<Integer> perm, List<List<Integer>> result, boolean[] used) {
        if (perm.size() == nums.length) {
            result.add(new ArrayList<>(perm));
            return;
        }
        for (int i = 0; i < nums.length; i++) {
            if (used[i]) {
                continue;
            }
            used[i] = true;
            //choose here
            perm.add(nums[i]);
            //backtrack
            backtrack(nums, perm, result, used);
            //remove chosen
            perm.remove(perm.size() - 1);
            used[i] = false;
        }
    }
}
